shell/common/isolate_configuration.h

```diff
@@ -19,30 +19,139 @@
+  /// @brief      Create a JIT isolate configuration using a list of futures to
+  ///             snapshots defining the ready isolate state. In environments
+  ///             where snapshot resolution in extremely expensive, embedders
+  ///             attempt to resolve snapshots on worker thread(s) and return
+  ///             the future of the promise of snapshot resolution to this
+  ///             method. That way, snapshot resolution begins well before
+  ///             isolate launch is attempted by the engine.
+  ///
+  /// @param[in]  kernel_pieces  The list of futures to Dart kernel snapshots.
+  ///
+  /// @return     A JIT isolate configuration.
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForKernelList(
       std::vector<std::future<std::unique_ptr<const fml::Mapping>>>
           kernel_pieces);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a JIT isolate configuration using the specified
+  ///             snapshot. This is a convenience method for the
+  ///             `CreateForKernelList` method that takes a list of futures to
+  ///             Dart kernel snapshots.
+  ///
+  /// @see        CreateForKernelList()
+  ///
+  /// @param[in]  kernel  The kernel shapshot.
+  ///
+  /// @return     A JIT isolate configuration.
+  ///
+  static std::unique_ptr<IsolateConfiguration> CreateForKernel(
+      std::unique_ptr<const fml::Mapping> kernel);
+
+  //----------------------------------------------------------------------------
+  /// @brief      Create a JIT isolate configuration using the specified
+  ///              snapshots. This is a convenience method for the
+  ///             `CreateForKernelList` method that takes a list of futures to
+  ///             Dart kernel snapshots.
+  ///
+  /// @see        CreateForKernelList()
+  ///
+  /// @param[in]  kernel_pieces  The kernel pieces
+  ///
+  /// @return     { description_of_the_return_value }
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForKernelList(
       std::vector<std::unique_ptr<const fml::Mapping>> kernel_pieces);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create an isolate configuration. This has no threading
+  /// restrictions.
+  ///
   IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Destroy an isolate configuration. This has no threading
+  ///             restrictions and may be collection of configurations may occur
+  ///             on any thread (and usually happens on an internal VM managed
+  ///             thread pool thread).
+  ///
   virtual ~IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      When an isolate is created and sufficiently initialized to
+  ///             move it into the `DartIsolate::Phase::LibrariesSetup` phase,
```

Comment:
Should these phases be in `|pipe|`s?

Reply:
I think both work but since I have not wired up the generator yet, I am not sure.